python/pyspark/ml/util.py

```diff
@@ -52,3 +71,133 @@ def _randomUID(cls):
         concatenates the class name, "_", and 12 random hex chars.
         """
         return cls.__name__ + "_" + uuid.uuid4().hex[12:]
+
+
+@inherit_doc
+class MLWriter(object):
+    """
+    .. note:: Experimental
+
+    Utility class that can save ML instances.
+
+    .. versionadded:: 2.0.0
+    """
+
+    def __init__(self, instance):
+        instance._transfer_params_to_java()
+        self._jwrite = instance._java_obj.write()
+
+    @since("2.0.0")
+    def save(self, path):
+        """Saves the ML instances to the input path."""
+        self._jwrite.save(path)
+
+    @since("2.0.0")
+    def overwrite(self):
+        """Overwrites if the output path already exists."""
+        self._jwrite.overwrite()
+        return self
+
+    @since("2.0.0")
+    def context(self, sqlContext):
+        """Sets the SQL context to use for saving."""
+        self._jwrite.context(sqlContext._ssql_ctx)
+        return self
+
+
+@inherit_doc
+class MLWritable(object):
+    """
+    .. note:: Experimental
+
+    Mixin for ML instances that provide MLWriter through their Scala
```

Comment:
Actually, this is a general mix-in; it does not rely on a Scala implementation.  It is MLWriter which uses the Scala implementation.